players/g1_player.py

```diff
@@ -286,36 +286,28 @@ def splash_zone_within_polygon(self, current_point: Tuple[float, float], target_
 
         distance = np.linalg.norm(np.array(current_point).astype(
             float) - np.array(target_point).astype(float))
+
         cx, cy = current_point
         tx, ty = target_point
         angle = np.arctan2(float(ty) - float(cy), float(tx) - float(cx))
         in_sandtrap = is_in_sand_trap(
             current_point, self.sand_trap_matlab_polys, cache=self.map_points_in_sand_trap)
-        splash_zone_poly_points = splash_zone(float(distance), float(
-            angle), float(conf), self.skill, current_point, in_sandtrap)
-        return self.shapely_poly.contains(ShapelyPolygon(splash_zone_poly_points))
 
-    def splash_zone_within_sand(self, current_point: Tuple[float, float], target_point: Tuple[float, float], conf: float) -> bool:
-        if type(current_point) == Point2D:
-            current_point = tuple(Point2D)
+        target_sandtrap = is_in_sand_trap(
+            target_point, self.sand_trap_matlab_polys, cache=self.map_points_in_sand_trap)
 
-        if type(target_point) == Point2D:
-            target_point = tuple(Point2D)
-
-        distance = np.linalg.norm(np.array(current_point).astype(
-            float) - np.array(target_point).astype(float))
-        cx, cy = current_point
-        tx, ty = target_point
-        angle = np.arctan2(float(ty) - float(cy), float(tx) - float(cx))
-        in_sandtrap = is_in_sand_trap(
-            current_point, self.sand_trap_matlab_polys, cache=self.map_points_in_sand_trap)
         splash_zone_poly_points = splash_zone(float(distance), float(
-            angle), float(conf), self.skill, current_point, in_sandtrap)
+            angle), float(conf), self.skill, current_point, in_sandtrap, target_sandtrap)
+
+        splash_in_poly = self.shapely_poly.contains(ShapelyPolygon(splash_zone_poly_points))
+
+        splash_in_sand = False
 
         for sand_trap in self.sand_trap_shapely_polys:
             if sand_trap.intersects(ShapelyPolygon(splash_zone_poly_points)):
-                return True
-        return False
+                splash_in_sand = True
```

Comment:
Once we find that the splash zone intersects with a sand trap, we can break the for loop already. No need to check if the splash zone intersects with any other sand traps.